src/demo/src/icecast/BufferArray.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @description Stores a collection of buffers as an array.
+ */
+export default class BufferArray {
```

Comment:
Probably don't need the array logic with the current MetadataPlayer implementation.